interfaces/connection.go

```diff
@@ -45,6 +46,39 @@ type ConnectedSlot struct {
 	dynamicAttrs map[string]interface{}
 }
 
+// AttrGetter is an interface with Attr getter method common
+// to ConnectedSlot, ConnectedPlug, PlugInfo and SlotInfo types.
+type AttrGetter interface {
+	Attr(key string) (interface{}, error)
+}
+
+func getAttribute(staticAttrs map[string]interface{}, dynamicAttrs map[string]interface{}, key string, val interface{}) error {
+	var v interface{}
+	var ok bool
+
+	if dynamicAttrs != nil {
+		v, ok = dynamicAttrs[key]
+	}
+	if !ok && staticAttrs != nil {
+		v, ok = staticAttrs[key]
+	}
+
+	if ok {
+		rt := reflect.TypeOf(val)
+		if rt.Kind() != reflect.Ptr || val == nil {
+			return fmt.Errorf("cannot store the value of attribute %q", key)
+		}
+
+		if reflect.TypeOf(v) != rt.Elem() {
+			return fmt.Errorf("the type of attribute %q is %s, expected %s", key, reflect.TypeOf(v).Name(), rt)
```

Comment:
Similarly, this message is going to surface to users due to errors from snap publishers. Neither of those need to be Go developers, so we don't want to expose strings such as _*map[string]interface{}_ in an error message.

Also, this is lacking some context that would enable to user or publisher to tell _where_ that attribute is.

Needs to be something similar to:

_"snap %q has interface %q with invalid value type for %q attribute"_

If you tweak the message further, let's please just preserve the hierarchy in the order of the arguments: snap > interface > attribute.